### PR TITLE
chore(skore): Remove train_test_split

### LINKED ISSUE #3028: Inconsistent behaviour between `train_test_split` and estimator's `checks`
### Describe the bug

Not sure if a bug but certainly an unexpected behavior: `skore.train_test_split` may raise a `HighClassImbalanceTooFewExamplesWarning` but check `[SKD004] High class imbalance` would be marked as "Passed".

I don't think the check should depend on the estimator, so I expect both objects to behave in a consistent way

### Steps/Code to Reproduce

```python
import pandas as pd
import skore
from sklearn.datasets import fetch_openml

german_credit = fetch_openml(data_id=31, as_frame=True, parser="pandas")
X, y = german_credit.data, german_credit.target

X_train, X_test, y_train, y_test = skore.train_test_split(
    X, y, random_state=0, test_size=0.2
)
```

raises a `HighClassImbalanceTooFewExamplesWarning`, but the check 
`[SKD004] High class imbalance` would be marked as "Passed":

```python
from sklearn.linear_model import LogisticRegression
from skrub import tabular_pipeline

tabular_pipeline_model = tabular_pipeline(LogisticRegression())
tabular_pipeline_report = skore.evaluate(tabular_pipeline_model, X, y)
tabular_pipeline_report.checks.summarize()
```

### Expected Behavior

Both `skore.train_test_split` and `skore.evaluate.checks` signal the class imbalance.

### Actual Behavior

They don't :)

### Environment

```shell
skore
-----
          pip: 25.0.1
        skore: 0.19.0
       jinja2: 3.1.6
       joblib: 1.4.2
   matplotlib: 3.10.1
        numpy: 2.2.5
       pandas: 2.2.3
       plotly: 5.24.1
         rich: 14.3.2
 scikit-learn: 1.8.0
      seaborn: 0.13.2
        skrub: 0.9.0
```

### PR DESCRIPTION
**Pull request by GaetandeCast**
Closes #3028

### CODE REVIEW

File: skore/src/skore/_sklearn/_cross_validation/report.py [hunk 1]
```diff
@@ -539,17 +539,17 @@ def create_estimator_report(
...
+        >>> linear_report = evaluate(
+        ...     LogisticRegression(random_state=42), X_train, y_train, splitter=5
         ... )
-        >>> forest_report = CrossValidationReport(
-        ...     RandomForestClassifier(random_state=42), X_train, y_train
+        >>> forest_report = evaluate(
+        ...     RandomForestClassifier(random_state=42), X_train, y_train, splitter=5
         ... )
-        >>> comparison_report = ComparisonReport([linear_report, forest_report])
+        >>> comparison_report = compare([linear_report, forest_report])
```
Comment (auguste-probabl) [inline skore/src/skore/_sklearn/_cross_validation/report.py]: a bit weird that this cross-validation method example relies on ComparisonReport

File: skore/src/skore/_sklearn/_comparison/inspection_accessor.py [hunk 2]
```diff
@@ -312,18 +277,13 @@ def permutation_importance(
...
-        6     big alpha        test      r2  ...     0.044482     0.013717
-        7     big alpha        test      r2  ...     0.088916     0.026191
-        8     big alpha        test      r2  ...     0.001826     0.008475
-        9     big alpha        test     mse  ...   288.618702    88.998809
-        10    big alpha        test     mse  ...   576.924696   169.940308
-        11    big alpha        test     mse  ...    11.846779    54.992092
+        0   small alpha        test      r2  ...     0.792711     0.029133
+        1   small alpha        test      r2  ...     1.582839     0.238285
+        2   small alpha        test      r2  ...     0.023727     0.005348
+        3   small alpha        test     mse  ...  -2663.409431   97.882726
```
Comment (auguste-probabl) [inline skore/src/skore/_sklearn/_comparison/inspection_accessor.py]: why is the MSE negative

Comment (GaetandeCast) [reply skore/src/skore/_sklearn/_comparison/inspection_accessor.py]: because `permutation_importance` expects a higher is better metric. Switched to `neg_mean_squared_error`.

**Review by auguste-probabl — APPROVED** [review]
LGTM, love to see it!
